src/middlewared/middlewared/plugins/sysdataset.py

```diff
@@ -100,6 +100,13 @@ async def do_update(self, job, data):
 
         verrors = ValidationErrors()
         if new['pool'] and new['pool'] != await self.middleware.call('boot.pool_name'):
```

Comment:
Seems like this check is not in the right place? For this error it should be new['pool'] != old['pool'] or something?

Reply:
@anodos325 ping. Just making sure it was not forgotten